yt/frontends/ahf/data_structures.py

```diff
@@ -0,0 +1,176 @@
+
+        self.n_ref = n_ref
+        self.over_refine_factor = over_refine_factor
+        super(AHFHalosDataset, self).__init__(
+            filename, dataset_type=dataset_type,
+            units_override=units_override, unit_system=unit_system
+        )
+
+    def _set_code_unit_attributes(self):
+        self.length_unit = self.quan(1.0, 'kpccm/h')
+        self.mass_unit = self.quan(1.0, 'Msun/h')
+        self.time_unit = self.quan(1.0, '')
+        self.velocity_unit = self.quan(1.0, 'km/s')
+
+    def _parse_parameter_file(self):
+        # This needs to set up the following items.  Note that these are all
+        # assumed to be in code units; domain_left_edge and domain_right_edge
+        # will be converted to YTArray automatically at a later time.
+        # This includes the cosmological parameters.
+        #
+        #   self.unique_identifier      <= unique identifier for the dataset
+        #                                  being read (e.g., UUID or ST_CTIME)
+        #   self.parameters             <= full of code-specific items of use
+        #   self.domain_left_edge       <= array of float64
+        #   self.domain_right_edge      <= array of float64
+        #   self.dimensionality         <= int
+        #   self.domain_dimensions      <= array of int64
+        #   self.periodicity            <= three-element tuple of booleans
+        #   self.current_time           <= simulation time in code units
+        #
+        # We also set up cosmological information.  Set these to zero if
+        # non-cosmological.
+        #
+        #   self.cosmological_simulation    <= int, 0 or 1
+        #   self.current_redshift           <= float
+        #   self.omega_lambda               <= float
+        #   self.omega_matter               <= float
+        #   self.hubble_constant            <= float
+
+        # Read all parameters.
+        simu = self._read_log_simu()
+        param = self._read_parameter()
+
+        # Set up general information.
+        self.filename_template = self.parameter_filename
+        self.file_count = 1
+        self.parameters.update(param)
+        self.particle_types = ('halos')
+        self.particle_types_raw = ('halos')
+        self.unique_identifier = \
+            int(os.stat(self.parameter_filename)[stat.ST_CTIME])
+
+        # Set up geometrical information.
+        self.refine_by = 2
+        self.dimensionality = 3
+        nz = 1 << self.over_refine_factor
+        self.domain_dimensions = np.ones(self.dimensionality, "int32") * nz
+        self.domain_left_edge = np.array([0.0, 0.0, 0.0])
+        # Note that boxsize is in Mpc but particle positions are in kpc.
+        self.domain_right_edge = np.array([simu['boxsize']] * 3) * 1000
+        self.periodicity = (True, True, True)
+
+        # Set up cosmological information.
+        self.cosmological_simulation = 1
+        self.current_redshift = param['z']
+        self.hubble_constant = param['Hubble(z)']
```

Comment:
I wrote a [post](https://groups.google.com/d/msg/ahfusers/PFkj8pA9cRM/4cj4SvNYBgAJ) in the AHFusers group and here's the response I got:

> Hi Bili, 
>
> there is absolutely no way to get h from AHF: when reading the simulation files (whatever format GADGET, TIPSY, DEVA, ART, etc.) units will be converted to Msun/h and Mpc/h and h is lost... 
>
> However, it is not difficult to simply drag this value from the simulation data through to maybe the logfile or the parameterfile; but that is not yet implemented, sorry. If urgently needed I will provide such a quick-and-dirty hack!? 
>
>bye, Alexander

Shall we ask for a hack? The problem with this solution is that all users need to use the hacked version of AHF in order to use the yt frontend. An alternative method would be to assign some nan value to `hubble_constant` and leave some notes in the doc. Any other suggestions?